roles/kubernetes/control_plane/init/primary/templates/kubeadm-config.j2

```diff
@@ -7,14 +7,21 @@
 #
 # kubeadm Configuration : https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta2/
 #
-{% if kubernetes_version is version('1.26.0', 'lt', version_type='semver') %}
-apiVersion: kubeadm.k8s.io/v1beta2
-{% else %}
+{% if kubernetes_version is version('1.31.0', 'lt', version_type='semver') %}
 apiVersion: kubeadm.k8s.io/v1beta3
+{% elif kubernetes_version is version('1.33.0', 'lt', version_type='semver') %}
+apiVersion: kubeadm.k8s.io/v1beta4
+{% else %}
+apiVersion: kubeadm.k8s.io/v1
```

Comment:
kubeadm only has v1beta3 and v1beta4 listed. There is no v1 listed at this time. As long as you're sure it will not break things if someone has entered the wrong k8s version this may be fine. I would only add these once available. Link to reference site for this repo - https://kubernetes.io/docs/reference/config-api/  Latest version of kubeadm config api - https://kubernetes.io/docs/reference/config-api/kubeadm-config.v1beta4/